### ISSUE TITLE
using the params arg in emayili::render()

### ISSUE THREAD

**Issue by yoav-raskin** [datawookie/emayili#79]
Hi,

I'm trying to write a workflow that automates emailing R markdown reports with filtered data. Each user should get his\her own unique filtered data fed into the email report.

My solution was to define a filter using an R markdown parameter, and then (for) loop over the `params` arg in the `render()` function, in the `emayili` R script. However this was not possible, since the `params` arg did not exist in the `emayili::render` function. 

Is there a workaround? Perhaps I could render outside `envelope()`, using `rmarkdown::render`, and then feed the outputted html(s) files into the `emayili::html()` function?

Thanks,
Yoav

**Comment by datawookie** [comment]
Hi @yoav-raskin,

Thanks for bringing this to my attention. I don't frequently use `params` in the YAML header, so this had not occurred to me.

However, I've added a `params` argument to `emayili::render()`. Would you mind testing it out for me? You can install the development version as follows:

```r
remotes::install_github("datawookie/emayili", ref = "8234e249277809248f51827f1bcb8b1c65eae12b")
```

Thanks, Andrew.

**Comment by yoav-raskin** [comment]
Hello @datawookie 

Thank you so much for the help!
I'm on it, will email you the feedback shortly!

Yoav

**Comment by yoav-raskin** [comment]
Hi @datawookie,

This is my code:


```
msg <- envelope() %>% 
  from("yoav.raskin@gmail.com") %>%
  to("yoav.r@kayma.com") %>% 
  subject("your class data is here!") %>%
  render("playground/emayili/class_report_emayili.Rmd", params = list(current_class = "D"))
smtp(msg, verbose = TRUE)
```

`current_class` is a parameter in the "playground/emayili/class_report_emayili.Rmd" file. This is the .Rmd header:


```
title: "emayili_class_report"
output:
  html_document: default
params:
  current_class: A
```

I'm getting an error - 
"Error: params object already exists in knit environment so can't be overwritten by render params "

Any thoughts?

**Comment by yoav-raskin** [comment]
@datawookie 

So the issue was that unlike` rmarkdown::render()`, the YAML header must **not** include a `params` argument.
When I remove it, `emayili::render()` defines and controls `params` in the .Rmd script, and it works perfectly.

**Comment by datawookie** [comment]
Aha! Okay, just pushed an update which should solve that issue. Please try:

```r
remotes::install_github("datawookie/emayili", ref = "962903b569a1e6717fad9fc0151060c452f82968")
```

**Comment by datawookie** [comment]
Hi @yoav-raskin, would you mind please testing this :point_up: and letting me know if it resolves the issue? Thanks, Andrew.

**Comment by yoav-raskin** [comment]
Hi @datawookie. Nothing seemed to be changed. 

This is me YAML:

`
title: "emayili_class_report"
output:
  html_document: default
params:
  current_class: A
`

I get: "Error: params object already exists in knit environment so can't be overwritten by render params "

**Comment by datawookie** [comment]
Hi @yoav-raskin, did you (1) install the package again using the command above (you need to use that command so that you get the right version!) and (2) restart your R session (so that the new version of the package is loaded). I'm 100% sure the issue is fixed... Thanks, Andrew.

**Comment by yoav-raskin** [comment]
Oh sorry, forgot to restart after installation. Works perfectly! 
Thank you very much!

Yoav

**Comment by datawookie** [comment]
Resolved in f2008e21a5d55700ad683f64349a3b6f8f538d51.
